includes/Modules/TagManager.php

```diff
@@ -342,12 +342,17 @@ protected function create_data_request( $method, $datapoint, array $data = array
 					$service = $this->get_service( 'tagmanager' );
 					return $service->accounts->listAccounts();
 				case 'list-containers':
-					if ( ! isset( $data['accountId'] ) ) {
-						/* translators: %s: Missing parameter name */
-						return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'accountId' ), array( 'status' => 400 ) );
-					}
-					$service = $this->get_service( 'tagmanager' );
-					return $service->accounts_containers->listAccountsContainers( "accounts/{$data['accountId']}" );
+					return function() use ( $data ) {
+						if ( ! isset( $data['accountId'] ) ) {
+							/* translators: %s: Missing parameter name */
+							return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'accountId' ), array( 'status' => 400 ) );
+						}
+						$service = $this->get_service( 'tagmanager' );
+						return array(
+							'containers' => $service->accounts_containers->listAccountsContainers( "accounts/{$data['accountId']}" ),
+							'accountId'  => $data['accountId'],
+						);
+					};
```

Comment:
There are two issues here:
* A Google API response should be modified in the `parse_data_response()` method, the `create_data_request()` method should always return a Google API request object when possible. Closures should only be returned for non Google API requests or very custom things.
* In this case here, I think the entire change is unnecessary and should be reverted - no point in returning an argument that was also passed via the request.